src/main/java/org/fcrepo/spec/testsuite/versioning/LdpvHttpGet.java

```diff
@@ -0,0 +1,343 @@
+            return Arrays.stream(x.getValue().split(","))
+                         .filter(val -> {
+                             return val.trim().equalsIgnoreCase("accept-datetime");
+                         }).count() > 0;
+        }).count(), 1, "Link header with " +
+                       "'Vary: accept-datetime' value " +
+                       "must be present");
+
+    }
+
+    private String convertToRfc1123DateTimeString(final String isoDateString) {
+        final DateTimeFormatter FMT = RFC_1123_DATE_TIME.withZone(ZoneId.of("UTC"));
+        return FMT.format(ISO_INSTANT.parse(isoDateString, Instant::from));
+    }
+
+    private Response createVersionedResource(final String uri, final TestInfo info) {
+        return createRequest().header("Link", ORIGINAL_RESOURCE_LINK_HEADER)
+                              .header(SLUG, info.getId())
+                              .when()
+                              .post(uri);
+    }
+
+    private void confirmPresenceOfTimeGateLink(final Response response) {
+        confirmPresenceOfRelType(response, "timegate");
+    }
+
+    private void confirmPresenceOfTimeMapLink(final Response response) {
+        confirmPresenceOfRelType(response, "timemap");
+    }
+
+    private void confirmPresenceOfRelType(final Response response, final String relType) {
+        Assert.assertEquals(getLinksOfRelType(response, relType).count(),
+                            1,
+                            "Link with rel type '" + relType + "' must be present but is not!");
+    }
+
+    private void confirmPresenceOfLinkValue(final String linkValue, final Response response) {
+        final Link link = Link.valueOf(linkValue);
+        final String relType = link.getRel();
+        Assert.assertEquals(getLinksOfRelType(response, link.getRel()).filter(l -> l.equals(link))
+                                                                      .count(),
+                            1,
+                            "Link header with a value of " + linkValue + " must be present but is not!");
+    }
+
+    private URI getTimeGateUri(final Response response) {
+        return getLinksOfRelTypeAsUris(response, "timegate").findFirst().get();
+    }
+
+    private URI getTimeMapUri(final Response response) {
+        return getLinksOfRelTypeAsUris(response, "timemap").findFirst().get();
+    }
+
+    private URI getOriginalUri(final Response response) {
+        return getLinksOfRelTypeAsUris(response, "original").findFirst().get();
+    }
+
+    private Stream<URI> getLinksOfRelTypeAsUris(final Response response, final String relType) {
+        return getLinksOfRelType(response, relType)
+            .map(link -> link.getUri());
+    }
+
+    private Stream<Header> getHeaders(final Response response, final String headerName) {
+        return response.getHeaders()
+                       .asList()
+                       .stream()
+                       .filter(header -> header.getName().equalsIgnoreCase(headerName));
```

Comment:
Instead of this filter, RestAssured provides:
http://static.javadoc.io/com.jayway.restassured/rest-assured/2.9.0/com/jayway/restassured/response/Headers.html#getList-java.lang.String-